### ISSUE TITLE
Loss will be fixed with 34.5388 with more than 12288 edges when training a GAE network

### ISSUE THREAD

**Issue by BillChow117** [pyg-team/pytorch_geometric#1185]
## To Reproduce

1. Training a GAE network with the number of edges (data.edge_index.size(1)) more than somewhere between 12288 and 16384, the loss will be fixed with 34.5388

```python
import torch
import torch.nn as nn
import torchvision
import torch.nn.functional as F
from torch_geometric.nn import GAE, VGAE, GCNConv, GraphConv
from torch_geometric.data import Data
from tqdm import trange

class Encoder1(nn.Module):
    def __init__(self, in_channels = None, out_channels = None):
        super(Encoder1, self).__init__()
        self.conv1 = GCNConv(in_channels,2 * out_channels, cached = True)
        self.conv2 = GCNConv(2 * out_channels, out_channels, cached = True)
    
    def forward(self, x, edge_index):        
        
        z = F.relu(self.conv1(x, edge_index))
        z = self.conv2(z, edge_index)   
        
        return z

class Encoder2(nn.Module):
    def __init__(self, in_channels = None, out_channels = None):
        super(Encoder2, self).__init__()
        self.conv1 = GraphConv(in_channels,2 * out_channels)
        self.conv2 = GraphConv(2 * out_channels, out_channels)
    
    def forward(self, x, edge_index):        
        
        z = F.relu(self.conv1(x, edge_index))
        z = self.conv2(z, edge_index)   
        return z


def get_model(lr = 0.001, GCN = 1, in_channels = None, Z_num = 16, device = 'cuda'):
    if GCN == 1:
        encoder = Encoder1(in_channels = in_channels, out_channels = Z_num)
    elif GCN == 2:
        encoder = Encoder2(in_channels = in_channels, out_channels = Z_num)

    model = GAE(encoder = encoder).to(device)
    optimizer = torch.optim.Adam(model.parameters(), lr = lr)

    return model, optimizer

def train(model, optimizer, data):
    model.train()
    optimizer.zero_grad()
    with torch.autograd.detect_anomaly():
        z = model.encode(data.x, data.edge_index)
        loss = model.recon_loss(z, data.edge_index)
        #loss = recon_loss + (1 / data.num_nodes) * model.kl_loss()
        loss.backward()
        optimizer.step()
    return loss

def get_Z(model, optimizer, epoch, data):
    model.reset_parameters()
    for i in trange(epoch):
        loss = train(model, optimizer, data)

    model.eval()
    z = model.encode(data.x, data.edge_index)
    return z, loss


device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
datapath = 'test/result/S_8/imagedata/000002.pt'
data = torch.load(datapath).to(device)
print(data)
inchannels = data.num_features
model, optimizer = get_model(in_channels = inchannels, device = device)
print(model)
print(optimizer)
Z, loss = get_Z(model, optimizer, 200, data)
print(Z, loss)
```
## Expected behavior
Loss doesn't changed

## Environment

* OS:ubuntu 18.04
* Python version:3.7.4
* PyTorch version:1.4.3
* CUDA/cuDNN version:10.1
* GCC version:7.4.0
* Any other relevant information:

**Comment by rusty1s** [comment]
So you are saying that the loss only stays the same in case the number of edges succeed a certain threshold? It's hard to believe that this is a PyG bug. It might instead be caused by your data. Is there a way for me to reproduce this error?

**Comment by ai-noahdolev** [comment]
I get exactly this phenomenon when running the tutorial using the Planetoid dataset.

**Comment by ai-noahdolev** [comment]
The solution is referenced in a different [issue](https://github.com/pyg-team/pytorch_geometric/issues/1165). Basically, you need to lower the LR.
